app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/TravelCertificateHistory.kt

```diff
@@ -0,0 +1,422 @@
+        onDismiss = {
+          reload()
+        },
+      )
+    }
+
+    CertificateHistoryUiState.FailureDownloadingHistory -> {
+      HedvigScaffold(
+        navigateUp = navigateUp,
+        modifier = Modifier.clearFocusOnTap(),
+        topAppBarText = stringResource(id = R.string.PROFILE_ROW_TRAVEL_CERTIFICATE),
+        topAppBarActions = {
+          IconButton(
+            onClick = { showBottomSheet = true },
+            modifier = Modifier.size(40.dp),
+          ) {
+            Icon(
+              imageVector = Icons.Hedvig.Info,
+              contentDescription = stringResource(R.string.REFERRALS_INFO_BUTTON_CONTENT_DESCRIPTION),
+              modifier = Modifier.size(24.dp),
+            )
+          }
+        },
+      ) {
+        Spacer(modifier = Modifier.weight(1f))
+        HedvigInformationSection(
+          title = stringResource(id = R.string.something_went_wrong),
+          buttonText = stringResource(id = R.string.GENERAL_RETRY),
+          onButtonClick = {
+            reload()
+          },
+        )
+        Spacer(modifier = Modifier.weight(1f))
+      }
+    }
+
+    CertificateHistoryUiState.Loading -> {
+      Box(modifier = Modifier.fillMaxSize()) {
+        CircularProgressIndicator(modifier = Modifier.align(Alignment.Center))
+      }
+    }
+
+    is CertificateHistoryUiState.SuccessDownloadingCertificate -> {
+      LaunchedEffect(uiState.downLoadingUri) {
+        uiState.downLoadingUri?.let {
+          onShareTravelCertificate(it)
+        }
+      }
+    }
+
+    is CertificateHistoryUiState.SuccessDownloadingHistory -> {
+      ShowHistory(
+        onIconClick = { showBottomSheet = true },
+        onCertificateClick = onCertificateClick,
+        onContinue = onContinue,
+        navigateUp = navigateUp,
+        historyList = uiState.certificateHistoryList,
+      )
+    }
+  }
+}
+
+@Composable
+private fun ShowHistory(
+  onIconClick: () -> Unit,
+  onCertificateClick: (String) -> Unit,
+  onContinue: () -> Unit,
+  navigateUp: () -> Unit,
+  historyList: List<TravelCertificate>,
+) {
+  HedvigScaffold(
+    navigateUp = navigateUp,
+    modifier = Modifier.clearFocusOnTap(),
+    topAppBarText = stringResource(id = R.string.PROFILE_ROW_TRAVEL_CERTIFICATE),
+    topAppBarActions = {
+      IconButton(
+        onClick = { onIconClick() },
+        modifier = Modifier.size(40.dp),
+      ) {
+        Icon(
+          imageVector = Icons.Hedvig.Info,
+          contentDescription = stringResource(R.string.REFERRALS_INFO_BUTTON_CONTENT_DESCRIPTION),
+          modifier = Modifier.size(24.dp),
+        )
+      }
+    },
+  ) {
+    if (historyList.isEmpty()) {
+      Spacer(modifier = Modifier.weight(1f))
+      ShowInitialInfo()
+    } else {
+      ShowNotEmptyList(
+        list = historyList,
+        onCertificateClick = onCertificateClick,
+      )
+    }
+    Spacer(modifier = Modifier.weight(1f))
```

Comment:
```suggestion
    Spacer(modifier = Modifier.weight(1f))
    Spacer(modifier = Modifier.height(16f))
```

You may have to do this here, so that if the parts above and the parts below are tall enough that there is no room for the weighted spacer to exist, then there should still be at least 16.dp of space (picked a random randomly now, maybe smth else makes more sense, depends on design) between those two